tsdb/store.go

```diff
@@ -252,6 +261,7 @@ func (s *Store) CreateShard(database, retentionPolicy string, shardID uint64) er
 	}
 
 	s.shards[shardID] = shard
+	s.shardLocations[shardID] = &shardLocation{Database: database, RetentionPolicy: retentionPolicy, Shard: shard}
```

Comment:
Any reason not to just combine `shards` and `shardLocations` maps?  `shardLocation` has a pointer to the `Shard` which is all that `shards` has.